Softeq.XToolkit.PushNotifications.iOS/IosPushNotificationsService.cs

```diff
@@ -0,0 +1,122 @@
+﻿// Developed by Softeq Development Corporation
+// http://www.softeq.com
+
+using System;
+using Softeq.XToolkit.Common.Interfaces;
+using UIKit;
+using UserNotifications;
+
+namespace Softeq.XToolkit.PushNotifications.iOS
+{
+    public class IosPushNotificationsService : PushNotificationsServiceBase
+    {
+        private readonly INotificationsPermissionsService _permissionsService;
+
+        private bool _isInitialized;
+
+        // badges
+
+        public IosPushNotificationsService(IRemotePushNotificationsService remotePushNotificationsService,
+            IPushTokenStorageService pushTokenStorageService, IPushNotificationsHandler pushNotificationsHandler,
+            IPushNotificationParser pushNotificationParser, INotificationsPermissionsService permissionsService, ILogManager logManager)
+            : base(remotePushNotificationsService, pushTokenStorageService, pushNotificationsHandler, pushNotificationParser, logManager)
+        {
+            _permissionsService = permissionsService;
+        }
+
+        public override void Initialize(bool showForegroundNotificationsInSystem)
+        {
+            if (_isInitialized)
+            {
+                Logger.Debug("PushNotificationsServiceIos: Already Initialized");
+                return;
+            }
+
+            _isInitialized = true;
+            UNUserNotificationCenter.Current.Delegate = new UserNotificationCenterDelegate(this, showForegroundNotificationsInSystem);
+        }
+
+        public override void RegisterForPushNotifications()
+        {
+            UIApplication.SharedApplication.InvokeOnMainThread(async () =>
+            {
+                if (UIApplication.SharedApplication.IsRegisteredForRemoteNotifications)
```

Comment:
what happened when you register push notification for the device, but server request failed?

Reply:
Right now it is not handled on library side but this can be handled on client side if you unregister from notifications in system when server registration fails.
This will be reworked

Reply:
Reworked